src/main/java/main/Main.java

```diff
@@ -0,0 +1,27 @@
+package main;
+
+import actions.BasicAction;
+import generationExamples.EquationPattern;
+
+import java.util.List;
+
+public class Main {
+    private static EquationPattern equationPattern;//STATIC - because it has an important data, that should be saved for whole runtime of app
```

Comment:
Ping me about that, I believe we could find few ways to get rid of this variable